### PR TITLE
feat: use Tokenizer class in swift-transformers to use BPE

### PR DESCRIPTION
**Pull request by ensan-hcl**
* The `Tokenizer` class in [swift-transformers](https://github.com/huggingface/swift-transformers) works greatly with GPT-2 tokenizers. I added a dependency to it and achieved correct BPE tokenization in Swift!
* I also updated the logic for decoding to get the correct behavior (output now has exactly 16 tokens as expected!
* I am observing `nan` output in logits so the model is not predicting as expected. This should be fixed in another PR.
* This PR requires https://github.com/Skyline-23/zenz-CoreML/pull/1 to be merged.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -6,11 +6,11 @@ import PackageDescription
...
     platforms: [
-        .iOS(.v15),
-        .macOS(.v12),
-        .macCatalyst(.v15),
-        .tvOS(.v15),
-        .watchOS(.v8),
+        .iOS(.v16),
+        .macOS(.v13),
+        .macCatalyst(.v16),
+        .tvOS(.v13),
```
Comment (Skyline-23) [inline Package.swift]: image

I think this tvOS version needs to be 15+. cause of ML model's minimum version.

File: Sources/swift-zenz-coreml/swift_zenz_coreml.swift [hunk 2]
```diff
@@ -1,80 +1,34 @@
...
 // Load the CoreML model
 func loadModel() -> zenz_v1? {
     let config = MLModelConfiguration()
     return try? zenz_v1(configuration: config)
 }
 
+// Load the Tokenizer model
+func loadTokenizer() async -> Tokenizer? {
+    do {
+        return try await AutoTokenizer.from(modelFolder: Bundle.module.resourceURL!)
```
Comment (Skyline-23) [inline Sources/swift-zenz-coreml/swift_zenz_coreml.swift]: Isn't it better use optional instead of force unwrapping?

File: Sources/swift-zenz-coreml/swift_zenz_coreml.swift [hunk 3]
```diff
@@ -107,29 +61,34 @@ func predict(text: String, model: zenz_v1, tokenizer: Tokenizer) -> [String] {
...
-    for i in 0..<logits.shape[1].intValue {
-        var logitValues = [Float]()
-        for j in 0..<logits.shape[2].intValue {
-            logitValues.append(logits[[0, i, j] as [NSNumber]].floatValue)
+    for batchID in 0..<logits.shape[0].intValue {
+        predictedTokenIDs.append([])
+        for i in 0..<logits.shape[1].intValue {
+            var logitValues = [Float]()
+            // get argMax
+            let maxId = (0..<6000).max {
```
Comment (Skyline-23) [inline Sources/swift-zenz-coreml/swift_zenz_coreml.swift:73]: 👍

**Review by Skyline-23 — APPROVED** [review]

File: Package.swift (hunk 1 above)
Comment (ensan-hcl) [reply Package.swift]: Changed to v16 based on the restriction in swift-transformer
